spec/buildkite_trigger_build_action_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'spec_helper'
+require 'buildkit'
+
+FakeResponse = Struct.new(:state, :web_url)
+
+describe Fastlane::Actions::BuildkiteTriggerBuildAction do
+  let(:buildkit) { Buildkit.new(token: 'test') }
+
+  before do
+    allow(Buildkit).to receive(:new).and_return(buildkit)
+  end
+
+  context 'when the API responds with state scheduled' do
+    it 'returns the new build web URL' do
+      expected_url = 'https://fake.url'
+      allow(buildkit).to receive(:create_build)
+        .and_return(FakeResponse.new('scheduled', expected_url))
```

Comment:
Wouldn't it be more idiomatic of `rspec` practices to return a `double(state: 'schedule', web_url: expected_url)` rather than creating our own `FakeResponse` type for the test?

Reply:
TIL about `double` and the [RuboCop-recommended](https://github.com/rubocop/rubocop-rspec/blob/863bafacd1a3313d707728fbba2623f8fcd9dd6b/lib/rubocop/cop/rspec/verified_doubles.rb#L6-L24) `instance_double`.

I left some notes about the resulting test implementation in 657c422. It's likely there's more Ruby/RSpec foo I'm not aware of that would address the minor issue I encountered there.

In the meantime, I'll merge the PR so the change is ready to ship in a new version. However, I won't be shipping a new version just yet. I have other unrelated things I'm working on and I might get to this later in the week. Just in case new PRs land in the meantime.